### DISCUSSION TITLE
or operator

### DISCUSSION THREAD

**Discussion by Inrap** [PostgREST/postgrest#2709] [Q&A] [answered]
### Environment

* PostgreSQL version: 14.5
* PostgREST version: 10
* Operating system: Windows Server
* Openlayers/css/html

### Description of issue

Following the [tutorial](https://postgrest.org/en/stable/api.html#operators), I try to use te `or` operator in a `fetch`.
I want to query 2 fields (ro, typope) from a PostgeSQL view. I tried : 

 `const test = fetch(url/ol_prescription?or=(ro.like.${ro},typope.like.${typope})` but it returns : 
`code: "PGRST100"
details: 'unexpected "=" expecting letter, digit, "-", "->>", "->" or delimiter (.)'`

So I tried to follow exactly the tutorial by imitating `GET /people?or=(age.lt.18,age.gt.21) HTTP/1.1` by replacing `age` with my primary key `gid`. It doesnt' work either : all the rows are returned i.e. no filter.

So I am confused, where is the error : bad use or doc ?
Thanks

**Comment by laurenceisla** [comment]
The GET example you gave should work, are you sure that all the rows are returned? i. e. including the ones which `gid`  values are `19` and `20`? 

For the `fetch` example, the `console.log()` of the exact generated URL that you're fetching would help. Or even better the whole fetch example.

Also, perhaps are you using Nginx or some other web server that changes the query parameters in some way?

**Comment by Inrap** [comment, marked as answer]
I answer step by step : 
Close to the `GET` tutorial example : 
`const test = fetch(url/ol_prescription?or=(gid.gt.18,gid.lt.100)
` 
returns all the rows of the view as shown in the console : 

![image](https://user-images.githubusercontent.com/32261793/225838101-47e5f36f-486b-4026-afc2-a493237515b5.png)

For my query with two fields : 
`const test = fetch(url/ol_prescription?or=(ro.like.${ro},typope.like.${typope})  .then(rep => rep.json()) .then(ope => {
      console.log(ope)
})`, the console returns : 

![image](https://user-images.githubusercontent.com/32261793/225839777-f3f98a84-3310-4400-974f-f424d44fdd6d.png)

If it can help : `const test = fetch(url/ol_prescription?ro=like.${ro}`)` works fine.

Apache is my web server.

**Reply by laurenceisla** [reply]
This request:  `url/ol_prescription?or=(gid.gt.18,gid.lt.100)` is telling PostgREST to query every `ol_prescription` with `gid` greater than 18 **OR** less than 100, which will result on all the rows to be selected due to the overlap. What's the result you want to get?

If you need to get the `gid` greater than 18 **AND** below 100 then adding both filters as query parameters is what you need:

```
url/ol_prescription?gid=gt.18&gid=lt.100
```

If you need all the `gid` that are below 18 or above 100 then:

```
url/ol_prescription?or=(gid.lt.18,gid.gt.100)
```

On your second query, the `,` in the filter that is separating the last name and name is the culprit. You'll need to wrap the filter in double quotes to fix it:

```
url/ol_prescription?or=(ro.like."${ro}",typope.like."${typope}"
```

**Reply by Inrap** [reply]
Perfectly clear. Thank you.
